src/lib/actions/create-activity.js

```diff
@@ -0,0 +1,42 @@
+const { isValidName } = require('../tools/validate');
+const { errors, parameters } = require('../../constants');
+
+function createActivity(params, activities) {
+  /* check request parameters */
+  if (typeof params.name !== 'string'
+    || params.name.length < parameters.name.min
+    || params.name.length > parameters.name.max
+  ) {
+    throw new Error(`${errors.common.INVALID_PARAMETER_VALUE}: --name`);
+  }
+  if (!isValidName(params.name)) {
+    throw new Error(errors.createActivity.INVALID_NAME);
+  }
+
+  /* execute action */
+  const match = activities.find(a => a.name === params.name);
+  if (match) {
+    return {
+      response: {
+        activityArn: match.activityArn,
+        creationDate: match.creationDate,
+      },
+    };
+  }
+
+  const activity = {
+    activityArn: `arn:aws:states:local:0123456789:activity:${params.name}`,
```

Comment:
- should use defined region
- accountId should be customizable (like for region, endpoint, ...) and we should use the value here